lib/client-api/src/types.ts

```diff
@@ -20,16 +24,24 @@ export interface StoryMetadata {
   decorators: DecoratorFunction[];
 }
 
-export interface StoreItem extends StoryContext {
-  id: string;
-  kind: string;
-  name: string;
+export interface AddStoryArgs extends StoryMetadata, StoryIdentifier {
+  id: StoryId;
```

Comment:
i think this is redundant since they are already declared in the parent interfaces

```
export type AddStoryArgs = StoryMetadata & StoryIdentifier & { storyFn: StoryFn };
```

Reply:
Yeah I don't get the difference between intersection types (`&`) and extending types. Someone who knows more about TS can explaining it to me? I think when you `extend` you have to "redundantly" name the fields; which is often quite useful though ;)

Reply:
Happy to be educated about TS conventions & best practices (cc @kroeder @gaetanmaisse), but without any further knowledge I'd say the DRY version is better.